lib/index.coffee

```diff
@@ -6,13 +6,13 @@ path = require 'path'
 # we need a blacklist rather than just a whitelist. this issue is caused by the
 # fact that roots doesn't re-run detection when a file's extension changes.
 BLACKLISTED_FILE_EXTENSIONS = [
-  '.jpg', '.gif', '.png', '.css', '.styl', '.js', '.coffee'
+  '.jpg', '.gif', '.png', '.css', '.styl', '.js', '.coffee', '.scss'
 ]
 
 module.exports = (data) ->
   class TumblrCompiler
     constructor: ->
-      @data = {}
+      @data = data || require './default'
```

Comment:
i liked:

```coffee
module.exports = (data) ->
  class TumblrCompiler
    data ?= require './default'

    fs: ->
      category: 'tumblr-markup'
      extract: true
      detect: (f) ->
        path.extname(f.relative) not in BLACKLISTED_FILE_EXTENSIONS

    compile_hooks: ->
      category: 'tumblr-markup'
      'after_file': (ctx) ->
        # just a crude extension check... the compiler will let the markup pass
        # through unchanged if it turns out this is just regular html
        if ctx['out_ext'] is 'html'
          ctx.content = compile(ctx.content.toString('utf8'), data)
```

i didn't know you could do the `?=` . the function wrapper seemed to be necessary.